bootstrap/cmd/bootstrap/app/server.go

```diff
@@ -534,19 +533,23 @@ func Run(opt *options.ServerOption) error {
 		// ks runApply API expects clientcmd.ClientConfig, which kind of have soft dependency on existence of ~/.kube/config
 		// if use k8s client-go API, would be quite verbose if we create all resources one by one.
 		// TODO: use API to create ks Components
-		log.Infof("Apply kubeflow Components...")
-		rawCmd := "ks apply default --token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)"
-		applyCmd := exec.Command("bash", "-c", rawCmd)
-
-		var out bytes.Buffer
-		var stderr bytes.Buffer
-		applyCmd.Stdout = &out
-		applyCmd.Stderr = &stderr
-		if err := applyCmd.Run(); err != nil {
-			log.Infof("stderr >>> " + fmt.Sprint(err) + ": " + stderr.String())
-			return err
-		} else {
-			log.Infof("Components applied: " + out.String())
+		for _, component := range bootConfig.App.Components {
+			log.Infof("Apply kubeflow component %v", component.Name)
+			rawCmd := fmt.Sprintf(
+				"ks apply default -c %v --token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)",
+				component.Name)
+			applyCmd := exec.Command("bash", "-c", rawCmd)
+
+			var out bytes.Buffer
+			var stderr bytes.Buffer
+			applyCmd.Stdout = &out
+			applyCmd.Stderr = &stderr
+			if err := applyCmd.Run(); err != nil {
```

Comment:
Can you print both stdout and stderr regardless of whether there was an error or not?

It helps understand what's going right/wrong

Reply:
updated